zebra-chain/src/primitives/viewing_key.rs

```diff
@@ -0,0 +1,119 @@
+//! Type definitions for viewing keys and their hashes.
+
+use std::fmt;
+
+use zcash_primitives::{sapling::SaplingIvk, zip32::DiversifiableFullViewingKey};
+
+use crate::parameters::Network;
+
+pub mod sapling;
+
+#[cfg(test)]
+mod tests;
+
+const KEY_HASH_BYTE_SIZE: usize = 32;
+
+#[derive(Debug, Clone)]
+/// A Zcash Sapling or Orchard viewing key
+// TODO: Add Orchard types and any other Sapling key types
```

Comment:
###### Optional Suggestion

Let's swap the attribute and comment as per https://github.com/rust-lang/style-team/issues/72.

```suggestion
/// A Zcash Sapling or Orchard viewing key
// TODO: Add Orchard types and any other Sapling key types
#[derive(Debug, Clone)]
```